pkg/memory-monitor/src/monitor/cgroups.c

```diff
@@ -0,0 +1,332 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/select.h>
+#include <syslog.h>
+#include <unistd.h>
+
+#include "config.h"
+#include "event.h"
+#include "monitor.h"
+#include "util.h"
+
+#include "cgroups.h"
+
+unsigned long cgroup_get_memory_usage(const char *cgroup_name) {
+    char path[PATH_MAX + 1];
+    int fd;
+    char str_usage[256];
+    unsigned long usage_bytes;
+
+    snprintf(path, sizeof(path), "%s/%s/memory.usage_in_bytes", CGROUP_PATH_PREFIX, cgroup_name);
+    fd = open(path, O_RDONLY);
+    if (fd == -1) {
+        syslog(LOG_ERR, "opening usage_in_bytes: %s", strerror(errno));
+        return 0;
+    }
+
+    ssize_t nread = read(fd, str_usage, sizeof(str_usage) - 1);
+    if (nread <= 0) {
+        syslog(LOG_ERR, "reading usage_in_bytes: %s", strerror(errno));
+        close(fd);
+        return 0;
+    }
+
+    str_usage[nread] = '\0';
+    bool error;
+    usage_bytes = strtoudec(str_usage, &error);
+    if (error) {
+        syslog(LOG_ERR, "strtoul: %s", strerror(errno));
+        close(fd);
+        return 0;
+    }
+
+    close(fd);
+    return usage_bytes;
+}
+
+
+unsigned long cgroup_get_total_cache(const char *cgroup_name) {
+    char buf[256];
+    FILE *file;
+    char line[256];
+    unsigned long total_cache = 0;
+
+    // Construct the path to the memory.stat file
+    snprintf(buf, sizeof(buf), "%s/%s/memory.stat", CGROUP_PATH_PREFIX, cgroup_name);
```

Comment:
You give `PATH_MAX` bytes (which is 4k?) in https://github.com/lf-edge/eve/pull/3984/files#diff-5840693cf44b592e67314a102e3c3d3a8a3a6e0fb488d13e832378741cf9b75cR29 but here only 256 bytes for the buffer.

Also you don't check the return value of `snprintf` ...

Reply:
Oops, yeah, I can replace it with PATH_MAX to be cleaner...

Reply:
Fixed. I also updated all the places where I work with char arrays and the snprintf function.